Update CMake files to support iOS builds

mpretty-cyro opened the pull request.

mpretty-cyro committed:
[WIP] Updated the StaticBuild.cmake to support mobile builds

mpretty-cyro committed:
[WIP] Tweaked StaticBuild.cmake so it compiles in both XCode and VSCode

mpretty-cyro committed:
Added missing Apple arch argument for multi-arch builds

mpretty-cyro committed:
Tweaks to build for the right platform and streamline build

mpretty-cyro committed:
[WIP] Latest tweaks, still have linker errors...

mpretty-cyro committed:
Manually linking all of the dependencies to get iOS build working

jagerman commented on cmake/StaticBuild.cmake:
```diff
@@ -259,38 +271,50 @@ if(APPLE AND CMAKE_CROSSCOMPILING)
-
 build_external(libtasn1
-    CONFIGURE_EXTRA --disable-doc
+    CONFIGURE_COMMAND ./configure ${gmp_build_host} --disable-shared --disable-doc --prefix=${DEPS_DESTDIR} --with-pic
```
We should probably rename `gmp_build_host` since it's now being used for other packages as well.

mpretty-cyro replied:
I've renamed it to `build_host` instead

mpretty-cyro committed:
Rename gmp_build_host to build_host (PR comments)

jagerman committed:
Move static lib deps behind libquic_internal

jagerman committed:
libtasn1: silence incredibly verbose warnings

GCC's analyzer produces some *massively* long warnings that produces
thousands of lines of build log spam.  This turns them off because we
don't care/can't do anything about them.

jagerman committed:
Update oxen-logging

To strip duplicates from source root.

jagerman committed:
Drop old "lokinet-ci-" docker image prefix

jagerman committed:
Remove obsolete/unused cmake/ngtcp2_lib.cmake

jagerman committed:
Fix hogweed target name in test suite
